segmentation_labeling_app/transforms/transformations.py

```diff
@@ -0,0 +1,212 @@
+                                video_array: np.ndarray):
+    """
+    Function to get video subset centered around coordinate pair with
+    size specified by box size tuple. Function takes a subset of each frame
+    and stacks together to get final video.
+    Args:
+        coordinate_pair: the coordinate on which to center the subset
+        box_size: the size of the subset box
+        video_array: the video to take the subset from
+
+    Returns:
+
+    """
+    transformed_coordinates = get_transformed_center(coordinate_pair, box_size,
+                                                     video_array.shape)
+    left_column = transformed_coordinates[0] - math.ceil(box_size[0] / 2)
+    right_column = transformed_coordinates[0] + math.ceil(box_size[0] / 2)
+    up_row = transformed_coordinates[1] - math.ceil(box_size[1] / 2)
+    down_row = transformed_coordinates[1] + math.ceil(box_size[1] / 2)
+    transformed_stack = []
+    for frame in video_array:
+        transformed_frame = frame[up_row:down_row, left_column:right_column]
+        transformed_stack.append(transformed_frame)
+    transformed_video = np.stack(transformed_stack)
+    return transformed_video
+
+
+def generate_max_ave_proj_image(video: np.ndarray, projection_type: str= 'average'):
+    """
+    Returns a maximum projection or an average projection of a video in
+    numpy array format
+    Args:
+        video: The video to generate the projection
+        projection_type: maximum or average, what type of projection
+
+    Returns: A numpy array generated with the specified strategy
+
+    """
+    if projection_type == 'average':
+        return np.mean(video)
+    elif projection_type == 'maximum':
+        return np.max(video, axis=0)
+
+
+def normalize_video(video: np.ndarray):
+    """
+    Function to normalize video by its global max and maximum 8 bit value
+    Args:
+        video: Video to be normalized
+    Returns:
+        norm_frames: normalized video
+    """
+    global_max = video.max()
+    norm_frames = []
+    for frame in video:
+        norm_frame = copy.deepcopy(frame)
+        norm_frame = np.uint8(norm_frame / global_max * 255)
+        norm_frames.append(norm_frame)
+    norm_frames = np.stack(norm_frames)
+    return norm_frames
+
+
+def transform_to_mp4(video: np.ndarray, output_path: str):
+    """
+    Function to transform 2p gray scale video into a mp4
+    video using opencv.
+    Args:
+        video: Video to be transformed
+        output_path: Output path for the transformed video
+
+    Returns:
+
+    """
+    writer = mpg.write_frames(output_path,
+                              video[0].shape,
+                              pix_fmt_in="gray",
+                              pix_fmt_out="gray",
+                              fps=30)
+    writer.send(None)
+    norm_video = normalize_video(video)
+    for frame in video:
+        frame = np.uint8(frame / frame.max() * 255)
```

Comment:
Shouldn't we use the global max here?

Reply:
No we don't need this it's an artifact from the previous testing I was doing, good catch.